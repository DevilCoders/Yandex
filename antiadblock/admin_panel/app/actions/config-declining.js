export const OPEN_CONFIG_DECLINING = 'OPEN_CONFIG_DECLINING';
export const CLOSE_CONFIG_DECLINING = 'CLOSE_CONFIG_DECLINING';

export function openConfigDeclining(serviceId, labelId, id) {
    return {
        type: OPEN_CONFIG_DECLINING,
        serviceId,
        labelId,
        id
    };
}

export function closeConfigDeclining(id) {
    return {
        type: CLOSE_CONFIG_DECLINING,
        id
    };
}

