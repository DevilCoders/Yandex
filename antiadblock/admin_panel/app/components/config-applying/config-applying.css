.config-applying
{
    &__copy-link-success
    {
        height: 28px;
        line-height: 28px;
    }

    &__copy-button
    {
        display: inline-block;
        margin-top: 15px;
    }
}
