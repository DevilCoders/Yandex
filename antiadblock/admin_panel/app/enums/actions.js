export const ACTIONS = {
    // Временно скрыт мониторинг на сервисах, т.к. он пока не работает
    // GRAPHS: 'graphs',
    CONFIGS: 'configs',
    AUDIT: 'audit',
    INFO: 'info',
    UTILS: 'utils',
    // ANTIADB-2636
    // HEALTH: 'health',
    RULES: 'rules',
    SBS_RESULT_CHECKS: 'sbs-result-checks'
};

export const SECRET_ACTIONS = {
    SCREENSHOTS: 'screenshots-checks',
    PROFILES: 'sbs-profiles',
    LOGS: 'logs'
};
