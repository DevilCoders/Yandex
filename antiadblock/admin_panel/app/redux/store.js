import {createStore, compose, applyMiddleware} from 'redux';
import reducer from 'app/reducers/index';
import thunkMiddleware from 'redux-thunk';
import {createLogger} from 'redux-logger';

let composeEnhancers = compose;

export function configureStore(state = {}) {
    if (typeof window !== 'undefined' && window.__REDUX_DEVTOOLS_EXTENSION__) {
        composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__;
    }
    if (process.env.NODE_ENV === 'local') {
        return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware, createLogger())));
    }
    return createStore(reducer, state, composeEnhancers(applyMiddleware(thunkMiddleware)));
}
