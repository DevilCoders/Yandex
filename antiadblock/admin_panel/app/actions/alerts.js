import servicesApi from 'app/api/services';
import {setErrors, clearErrors} from 'app/actions/errors';
import {getServices, getErrors} from 'app/reducers/index';
import {getById} from 'app/reducers/errors';
import {timeIsOver} from 'app/lib/errors-logic';

export const START_ALERTS_LOADING = 'START_ALERTS_LOADING';
export const END_ALERTS_LOADING = 'END_ALERTS_LOADING';

export function fetchAlerts() {
    return (dispatch, getState) => {
        dispatch(startAlertsLoading());

        return servicesApi.fetchAlerts().then(alerts => {
            const errors = getById(getErrors(getState()), 'dashboard');
            if (errors && errors.length) {
                dispatch(clearErrors('dashboard'));
            }

            return dispatch(endAlertsLoading(alerts.items));
        }, error => {
            const lastUpdateTime = getServices(getState()).lastUpdateTimeAlerts;

            if (timeIsOver(lastUpdateTime)) {
                return dispatch(setErrors('dashboard', [error.message]));
            }
        });
    };
}

export function startAlertsLoading() {
    return {
        type: START_ALERTS_LOADING
    };
}

export function endAlertsLoading(items) {
    return {
        type: END_ALERTS_LOADING,
        items
    };
}
