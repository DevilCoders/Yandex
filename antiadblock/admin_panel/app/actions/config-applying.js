import configApi from 'app/api/config';
import {setGlobalErrors} from 'app/actions/errors';

export const OPEN_CONFIG_APPLYING = 'OPEN_CONFIG_APPLYING';
export const CLOSE_CONFIG_APPLYING = 'CLOSE_CONFIG_APPLYING';
export const START_ACTIVE_CONFIG_LOADING = 'START_ACTIVE_CONFIG_LOADING';
export const END_ACTIVE_CONFIG_LOADING = 'END_ACTIVE_CONFIG_LOADING';

export function openConfigApplying(serviceId, id, configData, target) {
    return {
        type: OPEN_CONFIG_APPLYING,
        serviceId,
        id,
        configData,
        target
    };
}

export function closeConfigApplying(id) {
    return {
        type: CLOSE_CONFIG_APPLYING,
        id
    };
}

export function fetchCurrentConfig(labelId, target) {
    return dispatch => {
        dispatch(startActiveConfigLoading(labelId, target));

        return configApi.fetchCurrentConfig(labelId, target).then(config => {
            return dispatch(endActiveConfigLoading(labelId, target, config));
        }, error => {
            return dispatch(setGlobalErrors([error.message]));
        });
    };
}

export function startActiveConfigLoading(labelId, target) {
    return {
        type: START_ACTIVE_CONFIG_LOADING,
        labelId,
        target
    };
}

export function endActiveConfigLoading(labelId, target, config) {
    return {
        type: END_ACTIVE_CONFIG_LOADING,
        labelId,
        target,
        config
    };
}
