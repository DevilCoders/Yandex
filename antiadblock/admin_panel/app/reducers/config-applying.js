import * as configApplyingActions from 'app/actions/config-applying';
import update from 'app/lib/update';

export default function(state, action) {
    const {type} = action;
    const initialState = {
        visible: false,
        id: null,
        serviceId: null,
        target: null,
        configData: null,
        oldConfigData: null,
        loaded: false
    };

    state = state || initialState;

    switch (type) {
        case configApplyingActions.OPEN_CONFIG_APPLYING: {
            return {
                visible: true,
                id: action.id,
                serviceId: action.serviceId,
                target: action.target,
                configData: action.configData,
                oldConfigData: null,
                loaded: false
            };
        }
        case configApplyingActions.CLOSE_CONFIG_APPLYING: {
            return initialState;
        }
        case configApplyingActions.END_ACTIVE_CONFIG_LOADING: {
            return update(state, {
                $merge: {
                    oldConfigData: action.config,
                    loaded: true
                }
            });
        }
        default: {
            return state;
        }
    }
}
