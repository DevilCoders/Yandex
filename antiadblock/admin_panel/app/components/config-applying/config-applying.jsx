import React from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';

import YAML from 'js-yaml';
import {MonacoDiffEditor} from 'react-monaco-editor';

import Bem from 'app/components/bem/bem';

import Modal from 'lego-on-react/src/components/modal/modal.react';
import Button from 'lego-on-react/src/components/button/button.react';

import ModalClose from 'app/components/modal/__close/modal__close';
import LinkCopy from 'app/components/link-copy/link-copy';

import Preloader from '../preloader/preloader';
import {configType} from '../../types/index';

import {fetchCurrentConfig, closeConfigApplying} from 'app/actions/config-applying';
import {applyConfig} from 'app/actions/service';
import {getConfigApplying} from 'app/reducers/index';

import {KEYS} from 'app/enums/keys';

import i18n from 'app/lib/i18n';

import './config-applying.css';
import 'app/components/icon/_theme/icon_theme_link.css';

class ConfigApplying extends React.Component {
    constructor() {
        super();

        this.onApply = this.onApply.bind(this);
        this.onModalClose = this.onModalClose.bind(this);
        this.onKeyUp = this.onKeyUp.bind(this);
    }

    componentWillReceiveProps(nextProps) {
        if (!this.props.visible && nextProps.visible) {
            this.props.fetchCurrentConfig(nextProps.configData.label_id, nextProps.target);
            document.addEventListener('keyup', this.onKeyUp);
        } else if (this.props.visible && !nextProps.visible) {
            document.removeEventListener('keyup', this.onKeyUp);
        }
    }

    onApply() {
        this.props.applyConfig(this.props.configData.label_id, this.props.id, {
            target: this.props.target,
            oldConfigId: this.props.oldConfigData.id
        });
        this.props.closeConfigApplying(this.props.id);
    }

    onModalClose() {
        this.props.closeConfigApplying(this.props.id);
    }

    getLinkDiffConfigToCopyUrl() {
        const {serviceId, configData, oldConfigData} = this.props;

        return `https://${location.host}/service/${serviceId}/configs/diff/${oldConfigData.id}/${configData.id}`;
    }

    onKeyUp(e) {
        if (e.keyCode === KEYS.ENTER) {
            this.onApply();
        }
    }

    render() {
        return (
            <Modal
                autoclosable
                onDocKeyDown={this.onModalClose}
                onOutsideClick={this.onModalClose}
                visible={this.props.visible}>
                {!this.props.loaded ?
                    <Preloader /> :
                    [
                        <Bem
                            key='title'
                            block='modal'
                            elem='title'>
                            {i18n('service-page', 'applying-title')}
                            &nbsp;
                            <LinkCopy
                                text={i18n('hints', 'link-copied-to-buffer')}
                                to='right'
                                value={this.getLinkDiffConfigToCopyUrl()} />
                        </Bem>,
                        <ModalClose
                            key='close'
                            onClick={this.onModalClose} />,
                        <Bem
                            key='body'
                            block='modal'
                            elem='body'>
                            <MonacoDiffEditor
                                language='yaml'
                                options={{
                                    readOnly: true,
                                    scrollBeyondLastLine: false,
                                    minimap: {
                                        enabled: false
                                    }
                                }}
                                original={YAML.safeDump(this.props.oldConfigData.data, {
                                    lineWidth: Infinity
                                })}
                                value={YAML.safeDump(this.props.configData.data, {
                                    lineWidth: Infinity
                                })} />
                        </Bem>,
                        <Bem
                            key='actions'
                            block='modal'
                            elem='actions'>
                            <Button
                                theme='action'
                                view='default'
                                tone='grey'
                                size='s'
                                mix={{
                                    block: 'modal',
                                    elem: 'action'
                                }}
                                onClick={this.onApply}>
                                {i18n('common', 'apply')}
                            </Button>
                            <Button
                                theme='normal'
                                view='default'
                                tone='grey'
                                size='s'
                                mix={{
                                    block: 'modal',
                                    elem: 'action'
                                }}
                                onClick={this.onModalClose}>
                                {i18n('common', 'cancel')}
                            </Button>
                        </Bem>
                    ]}
            </Modal>
        );
    }
}

ConfigApplying.propTypes = {
    id: PropTypes.number,
    visible: PropTypes.bool,
    serviceId: PropTypes.string,
    target: PropTypes.string,
    configData: configType,
    oldConfigData: configType,
    loaded: PropTypes.bool,
    fetchCurrentConfig: PropTypes.func.isRequired,
    applyConfig: PropTypes.func.isRequired,
    closeConfigApplying: PropTypes.func.isRequired
};

export default connect(state => {
    return {
        id: getConfigApplying(state).id,
        visible: getConfigApplying(state).visible,
        serviceId: getConfigApplying(state).serviceId,
        target: getConfigApplying(state).target,
        configData: getConfigApplying(state).configData,
        oldConfigData: getConfigApplying(state).oldConfigData,
        loaded: getConfigApplying(state).loaded
    };
}, dispatch => {
    return {
        fetchCurrentConfig: (labelId, target) => {
            dispatch(fetchCurrentConfig(labelId, target));
        },
        applyConfig: (labelId, configId, target) => {
            dispatch(applyConfig(labelId, configId, target));
        },
        closeConfigApplying: id => {
            dispatch(closeConfigApplying(id));
        }
    };
})(ConfigApplying);
