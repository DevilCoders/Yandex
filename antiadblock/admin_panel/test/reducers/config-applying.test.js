import reducer from 'app/reducers/config-applying';
import * as configApplyingActionFixtures from '../fixtures/actions/config-applying';
import {CONFIG_APPLYING_EMPTY, CONFIG_APPLYING_OPENED, CONFIG_APPLYING_LOADED} from '../fixtures/store';

describe('config-applying reducer', () => {
    const initialState = CONFIG_APPLYING_EMPTY;
    const openedState = CONFIG_APPLYING_OPENED;
    const loadedState = CONFIG_APPLYING_LOADED;

    it('should return the initial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    });

    it('should handle OPEN_CONFIG_APPLYING', () => {
        expect(
            reducer(initialState, configApplyingActionFixtures.OPEN_CONFIG_APPLYING)
        ).toMatchSnapshot();
    });

    it('should handle START_ACTIVE_CONFIG_LOADING', () => {
        expect(
            reducer(openedState, configApplyingActionFixtures.START_ACTIVE_CONFIG_LOADING)
        ).toMatchSnapshot();
    });

    it('should handle END_ACTIVE_CONFIG_LOADING', () => {
        expect(
            reducer(loadedState, configApplyingActionFixtures.END_ACTIVE_CONFIG_LOADING)
        ).toMatchSnapshot();
    });

    it('should handle CLOSE_CONFIG_PREVIEW', () => {
        expect(
            reducer(loadedState, configApplyingActionFixtures.CLOSE_CONFIG_APPLYING)
        ).toMatchSnapshot();
    });
});
