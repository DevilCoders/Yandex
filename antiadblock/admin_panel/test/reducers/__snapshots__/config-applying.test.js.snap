// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config-applying reducer should handle CLOSE_CONFIG_PREVIEW 1`] = `
Object {
  "configData": null,
  "id": null,
  "loaded": false,
  "oldConfigData": null,
  "serviceId": null,
  "target": null,
  "visible": false,
}
`;

exports[`config-applying reducer should handle END_ACTIVE_CONFIG_LOADING 1`] = `
Object {
  "activeConfigData": Object {
    "comment": "made a config",
    "date": "2018-01-10T11:12:13",
    "id": 234,
    "statuses": Array [
      Object {
        "comment": "",
        "status": "active",
      },
    ],
  },
  "configData": Object {
    "comment": "made a config",
    "data": Object {
      "field1": "value1",
      "field2": "value2",
      "field3": "value3",
    },
    "date": "2018-01-10T11:12:13",
    "id": 236,
    "statuses": Array [
      Object {
        "comment": "",
        "status": "preview",
      },
    ],
  },
  "id": 236,
  "loaded": true,
  "oldConfigData": Object {
    "comment": "made a config",
    "data": Object {
      "field1": "value1",
      "field2": "value2",
      "field3": "value3",
    },
    "date": "2018-01-10T11:12:13",
    "id": 234,
    "statuses": Array [
      Object {
        "comment": "",
        "status": "active",
      },
    ],
  },
  "serviceId": 123,
  "visible": true,
}
`;

exports[`config-applying reducer should handle OPEN_CONFIG_APPLYING 1`] = `
Object {
  "configData": Object {
    "comment": "made a config",
    "data": Object {
      "field1": "value1",
      "field2": "value2",
      "field3": "value3",
    },
    "date": "2018-01-10T11:12:13",
    "id": 236,
    "statuses": Array [
      Object {
        "comment": "",
        "status": "preview",
      },
    ],
  },
  "id": 236,
  "loaded": false,
  "oldConfigData": null,
  "serviceId": 123,
  "target": "active",
  "visible": true,
}
`;

exports[`config-applying reducer should handle START_ACTIVE_CONFIG_LOADING 1`] = `
Object {
  "configData": Object {
    "comment": "made a config",
    "data": Object {
      "field1": "value1",
      "field2": "value2",
      "field3": "value3",
    },
    "date": "2018-01-10T11:12:13",
    "id": 236,
    "statuses": Array [
      Object {
        "comment": "",
        "status": "preview",
      },
    ],
  },
  "id": 236,
  "serviceId": 123,
  "type": "OPEN_CONFIG_APPLYING",
}
`;

exports[`config-applying reducer should return the initial state 1`] = `
Object {
  "configData": null,
  "id": null,
  "loaded": false,
  "oldConfigData": null,
  "serviceId": null,
  "target": null,
  "visible": false,
}
`;
